sway-core/src/type_system/binding.rs

```diff
@@ -73,6 +73,7 @@ use super::{TypeArgument, TypeId};
 pub struct TypeBinding<T> {
     pub inner: T,
     pub type_arguments: Vec<TypeArgument>,
+    pub prefix_type_arguments: Vec<TypeArgument>,
```

Comment:
As far as I understand (please correct me if I'm wrong), `type_arguments` and `prefix_type_arguments` cannot be both non-empty. If that's the case, can we then use a different representation that ensures this by construction? Something isomorphic to

```
enum TypeArgs {
    Regular(Vec<TypeArgument>),
    Prefix(Vec<TypeArgument>),
}

pub struct TypeBinding<T> {
...
    pub type_arguments: TypeArgs,
...
}
```

perhaps?

Reply:
Thanks for the review, most recent version applies this suggestion.